### PR TITLE
Update mapbox-gl to 1.0.0

### PR DESCRIPTION
**Pull request by hostr**
Supports new Mapbox "map views" pricing model
https://github.com/mapbox/mapbox-gl-js/releases

### CODE REVIEW

**Comment by Pessimistress** [comment]
Is mapbox continuing to maintain the 0.x track? Would someone want to stay on the old pricing model?

**Comment by hostr** [comment]
According to the [Mapbox blog bost](https://blog.mapbox.com/new-pricing-46b7c26166e7) 75% of users will save money by switching to the new map load pricing model. The other 25% will have until November 20th to make the switch and may want to wait the full time until upgrading to v1.0.0. 

I've submitted a support ticket asking about the 0.x support and backwards compatibility.

**Comment by ibgreen** [comment]
Should we also do a major version bump to make sure our users don't quietly get opted in to a new pricing model?

**Comment by Pessimistress** [comment]
If we are doing a major version bump we may consider making mapbox-gl a peerDependency? We are decoupled enough at this point we may not require a specific version.

**Comment by hostr** [comment]
Following up on this, I got word back from Mapbox support -
* The v0.xx will not be updated anymore
* Users can continue to use v0.54.xx after Nov 20th and it will use [vector tile pricing](https://www.mapbox.com/pricing/#vectortile)
* You cannot use v1.xx.xx and have the old pricing (map views)

A major version bump is probably a good idea.

File: package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "mapbox-gl": "~0.53.0",
+    "mapbox-gl": "~1.0.0",
```
Comment (Pessimistress) [inline package.json]: We will go with a new major release. Let's relax it to `^1.0.0`.

Comment (hostr) [reply package.json]: Updated!

**Comment by Pessimistress** [comment]
Published 5.0.0

**Pull request merged**